src/NexusMods.App.UI/Controls/Node.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections.ObjectModel;
+using Avalonia.Controls.Models.TreeDataGrid;
+using JetBrains.Annotations;
+using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+
+namespace NexusMods.App.UI.Controls;
+
+/// <summary>
+/// Base class for models displayed in a <see cref="Avalonia.Controls.TreeDataGrid"/>.
+/// </summary>
+[PublicAPI]
+public class Node<TNode> : ReactiveObject, IDisposable
```

Comment:
I would rename to be more descriptive, as node is a bit too generic. 
`TreeDataGridNode` maybe